pkg/krbd/device.go

```diff
@@ -76,13 +76,16 @@ func (d *Device) readDeviceAttrs(path string) error {
 
 	// Iterate over all available struct fields
 	for i := 0; i < t.NumField(); i++ {
-		tag := t.Field(i).Tag.Get("krbd")
-		if tag != "" {
-			r, err := os.Open(path + "/" + tag)
-			defer r.Close()
+		tag := getDeviceTag(t.Field(i))
+		if tag.name != "" {
+			r, err := os.Open(path + "/" + tag.name)
 			if err != nil {
+				if tag.optional {
+					continue
```

Comment:
Perhaps we should only allow `errors.Is(err, os.ErrNotExist)` if the tag is optional, so down the road we don't miss passing up any odd errors when reading these optional attributes.

Reply:
Seems like a good idea.

New push uses:
```go
if tag.optional && errors.Is(err, os.ErrNotExist) {
```